ssl/ssl_lib.c

```diff
@@ -4209,11 +4209,45 @@ int SSL_CTX_set_default_verify_file(SSL_CTX *ctx)
     return 1;
 }
 
+int SSL_CTX_set_default_verify_store(SSL_CTX *ctx)
+{
+    X509_LOOKUP *lookup;
+
+    lookup = X509_STORE_add_lookup(ctx->cert_store, X509_LOOKUP_store());
+    if (lookup == NULL)
+        return 0;
+
+    X509_LOOKUP_add_store(lookup, NULL);
+
+    /* Clear any errors if the default file does not exist */
+    ERR_clear_error();
```

Comment:
Perhaps `ERR_set_mark()` and `ERR_pop_to_mark()`?

Reply:
I'm wondering why.  Why should we preserve a *previous* error record stack that has nothing to do with this function.  I see no value in that, and rather see that as possibly confusing.  If anyone cares about failures, they normally care about the error information that's relevant for *that* failure, not the previous ones.

It's a little like *errno*...  there's no guarantee that it gets preserved by calls that affect it, no matter what.  Nor should there be.

Reply:
Well, I guess mainly because its an unexpected side effect which is generally considered undesirable.  You wouldn't expect calling this function to have the side effect of clearing the error queue.

Reply:
I still feel this isn't correct

Reply:
I would actually expect that *anything* can happen to the error queue, and that its content is only valid on failure.  However, because it *is* a queue, it's really impossible to reset it except by clearing it on success.

But tell you what...  I don't agree with your point of view, but if mark-n-pop makes you feel better, I see no reason for me to be hard-headed about this, it's not *that* important.  Choosing my fights 'n all that, you know?  :wink:

Reply:
.....

Should I change `SSL_CTX_set_default_verify_dir()` and `SSL_CTX_set_default_verify_file()` too?

Reply:
I ended up doing so.

Reply:
Thanks (and the answer would have been yes)